lib/Service/Social/FacebookProvider.php

```diff
@@ -59,25 +59,54 @@ public function supportsContact(array $contact):bool {
 	public function getImageUrls(array $contact):array {
 		$profileIds = $this->getProfileIds($contact);
 		$urls = [];
+
 		foreach ($profileIds as $profileId) {
-			$recipe = 'https://graph.facebook.com/{socialId}/picture?width=720';
-			$connector = str_replace("{socialId}", $profileId, $recipe);
-			$urls[] = $connector;
+			$url = $this->getImageUrl($profileId);
+			if (isset($url)) {
+				$urls[] = $url;
+			}
 		}
+
 		return $urls;
 	}
 
+	/**
+	 * Returns the profile-picture url
+	 *
+	 * @param {string} profile url
+	 *
+	 * @return string|null
+	 */
+	protected function getImageUrl(string $profileUrl):?string {
+		try {
+			$result = $this->httpClient->get($profileUrl);
+			$htmlResult = $result->getBody();
+
+			$avatar = '/meta property="og:image" content="(http[^"]+)/';
```

Comment:
the layout changed and I cannot figure out how to get the picture. but it seems still to be available without login.